Communitter/src/main/java/com/communitter/api/service/CommentService.java

```diff
@@ -0,0 +1,53 @@
+package com.communitter.api.service;
+
+import java.util.Date;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import com.communitter.api.dto.CommentDto;
+import com.communitter.api.model.Comment;
+import com.communitter.api.model.Post;
+import com.communitter.api.model.User;
+import com.communitter.api.repository.CommentRepository;
+import com.communitter.api.repository.PostRepository;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final CommentRepository commentRepository;
+    private final PostRepository postRepository;
+    public Logger logger = LoggerFactory.getLogger(JwtService.class);
+
+    @Transactional
+    public CommentDto createComment(Long postId, CommentDto commentDto){
+        Post post = postRepository.findById(postId).orElseThrow();
+        User author= (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Comment createdComment = Comment.builder().date(new Date())        
+        .post(post)              
+        .author(author)          
+        .content(commentDto.getContent()) 
+        .build();
+
+        logger.info("Created Comment: ");
+        logger.info("ID: {}", createdComment.getId());
+        logger.info("Content: {}", createdComment.getContent());
+        logger.info("Date: {}", createdComment.getDate());
+        logger.info("Post ID: {}", createdComment.getPost().getId());
+        logger.info("Author ID: {}", createdComment.getAuthor().getId());
+        commentRepository.save(createdComment);
+        return commentDto;
+    }
+
+    @Transactional
+    public String deleteComment(Long id){
+        Comment comment = commentRepository.findById(id).orElseThrow();
```

Comment:
comments can be deleted either by the person who created it or the owner. Where do you do this check?

Reply:
Also can we add try catch here as well

Reply:
> comments can be deleted either by the person who created it or the owner. Where do you do this check?

![image](https://github.com/user-attachments/assets/1f40e604-2dcb-49c3-be92-ce58f10dc53d)

Reply:
I will be adding those backend checkers once I engage editing tasks